Week2/todo_app/program.js

```diff
@@ -16,43 +16,139 @@ class TodoModel {
                 callback(err);
                 return;
             }
-
             callback(null, results);
         });
     }
 
-    create(description, callback) {
+    create(description, user_id, callback) {
         // Write code and query to create a new TODO item
+        const createQuery = `
+        INSERT INTO todo_items(text, is_completed, user_id) VALUES("${description}" , 0 , ${user_id})
+        `;
+        this.dbConnection.query(createQuery, (err,results, fields) => {
+            if(err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results);
+            }
+        })
+
     }
 
     update(id, description, callback) {
         // Write code and query to update and existing TODO item
+        const updateQuery = `
+        UPDATE todo_items SET text = "${description}" 
+        WHERE id = ${id}
+        `
+        ;
+        this.dbConnection.query(updateQuery, (err,results, fields) => {
+            if(err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results);
+            }
+        })
+
     }
 
     delete(id, callback) {
         // Write code and query to delete an existing TODO item
+        const deleteQuery = `
+        DELETE FROM todo_items WHERE id = ${id}
+        `
+        ;
+        this.dbConnection.query(deleteQuery, (err,results, fields) => {
+            if(err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results);
+            }
+        })
+
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
         // Write code and query add a tag to a TODO item
+        const tagTodoItemQuery = `
+        INSERT INTO todo_item_tag(todo_item_id, tag_id)
+        VALUES (${todoItemId},${tagId})
+        `
+        ;
+        this.dbConnection.query(tagTodoItemQuery, (err,results, fields) => {
+            if(err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results);
+            }
+        })
     }
         
     untagTodoItem(todoItemId, tagId, callback) {
         // Write code and query remove a tag from a TODO item
+        const untagTodoItemQuery = `
+        DELETE FROM todo_item_tag
+        WHERE todo_item_id = ${todoItemId}
```

Comment:
This is not correct, it will remove all tags for that item. Remember that tags are many-to-many - one item can have many tags, and one tag can be assigned to many items.

Try and improve this so that it only removes the specified tag for the specified item.